### DISCUSSION TITLE
Failed to Fetch Content - /api/framework/v1/read/NCF

### DISCUSSION THREAD

**Discussion by ruchinksinha** [Sunbird-Ed/Community#461] [Installation]
Hi,

The content load is failing. Please help in resolving the issue. Below are the details:

**URL:** https://<domain_name>/api/framework/v1/read/NCF?categories=board,gradeLevel,medium,class,subject

**Response:**
{
    "id": "api.framework.read",
    "ver": "1.0",
    "ts": "2023-04-13T18:33:36.971Z",
    "params": {
        "resmsgid": "b42ce9b0-da29-11ed-9a8e-d53b9bbefcf9",
        "msgid": null,
        "status": "failed",
        "err": null,
        "errmsg": null
    },
    "responseCode": "SERVER_ERROR",
    "result": {}
}

@swayangjit @luckynvdu - Please suggest the issue.

**Comment by chitranshu-keshav** [comment]
@ruchinksinha The information provided in the thread is not sufficient to provide assistance on issue that you have reported. Can you please provide additional detail so that the community can help you out with resolving the issue. Some examples of additional details are below

1. Issue is being faced in which version of Sunbird ED?
2. Is this issue faced only with a specific content or all content of the content type?
3. Is it being faced in all the mobile devices or a specific mobile device ?
4. Is it happening for a specific version of mobile OS ?
5. In which version of ED mobile app is this issue replicated?
6. Share the logs and the analysis done so far by your team and call out what specific help you need.

**Comment by ruchinksinha** [comment]
Hi, Please find my answers below:

1) Version 5.1.0
2) On all contents
3) On Web
4) On Web
5) On Web

**Reply by chitranshu-keshav** [reply]
@ruchinksinha can you share the analysis that you have done? Also share the logs etc. for community to get back  

@rajeevsathish can you please check on this with

**Reply by swayangjit** [reply]
@ruchinksinha  Please let us know if [learning platform](https://github.com/Sunbird-Knowlg/sunbird-learning-platform) is installed properly or not . If yes then share the log of errors when you are calling the API.

**Reply by ruchinksinha** [reply]
@swayangjit All the jenkins pipeline as per the documentation in https://ed.sunbird.org/use/installing-ed-platform/release-5.1.0 were successful. Can you elaborate on which log shall I provide for further investigate the framework read issue.

**Reply by ruchinksinha** [reply]
@swayangjit I can see the below error in the knowledgemw pod.

 [ERROR] api - {"id":"07530b15-2c52-0829-5ab0-211e71e15592","msg":"Getting error from ekstep while fetching framework by id ","additionalInfo":{"frameworkId":"NCF","queryString":""},"err":{"err":null,"responseCode":"SERVER_ERROR"}}

**Reply by swayangjit** [reply]
@ruchinksinha  I have created an issue in [Knowldg discussion forum](https://github.com/orgs/Sunbird-Knowlg/discussions/124). Please follow that.

**Comment by rajeevsathish** [comment]
@ruchinksinha hope the creation of default framework also done 
<img width="889" alt="Screenshot 2023-04-20 at 7 46 25 PM" src="https://user-images.githubusercontent.com/5653717/233394334-ac2c7c9e-0264-4711-b460-1aad30a24ee5.png">
if this is done please check in the db it is there.

**Comment by ruchinksinha** [comment]
@rajeevsathish During the execution of the PostInstallScript, the job had failed multiple time and I tried to resolve all the issues. finally the job was successful. Now since the job was executed multiple time, its showing my as below in the application.

![image](https://user-images.githubusercontent.com/10114264/233536757-197bbf2b-1902-498c-9cf3-61b9bfb56df7.png)

I tried to retire all the existing frameworks using the retire API but its still showing multiple. So I wanted to cleanup all the framework and create fresh one. But now I am unable to create new framework because of the below error.

**Request:**
{
    "request": {
        "framework": {
            "name": "NCF Demo",
            "code": "ncfdemo",
            "description": "NCF Demo Framework"
        }
    }
}

**Response:**
{
    "id": "api.framework.create",
    "ver": "1.0",
    "ts": "2023-04-21T03:52:59.272Z",
    "params": {
        "resmsgid": "01c1e080-dff8-11ed-af80-afa5f3a40bf4",
        "msgid": null,
        "status": "failed",
        "err": null,
        "errmsg": null
    },
    "responseCode": "SERVER_ERROR",
    "result": {}
}

Error in the knowledgemw corresponding to this API call is below:
[ERROR] api - {"id":"01c14440-dff8-11ed-a984-a754a0eac4d5","msg":"Getting error from ekstep while framework creation","additionalInfo":{"ekStepReqData":{"request":{"framework":{"name":"NCF Demo","code":"ncfdemo","description":"NCF Demo Framework"}}}},"err":{"err":null,"responseCode":"SERVER_ERROR"}}

**Reply by rajeevsathish** [reply]
@ruchinksinha can you please share logs from Learning-service which is installed in your VM so that we can see what is the issue you are facing from the upstream.

cc: @chitranshu-keshav @Krishnaj20 @vinukumar-vs

**Reply by ruchinksinha** [reply]
@rajeevsathish Are you referring to the tomcat catalina logs?

**Reply by ruchinksinha** [reply]
@rajeevsathish There are no errors in the catalina logs in the folder /home/learning/apache-tomcat-8.0.36/logs/ on the VM where learning-service is installed. Also the output of health check is below:

curl -sS localhost:8080/learning-service/health

{
  "id": "learning-service.health",
  "ver": "1.0",
  "ts": "2023-04-25T08:47:41ZZ",
  "params": {
    "resmsgid": "34373cbf-acb8-402a-bd99-8ffcc4552795",
    "msgid": null,
    "err": null,
    "status": "successful",
    "errmsg": null
  },
  "responseCode": "OK",
  "result": {
    "checks": [
      {
        "healthy": true,
        "name": "domain graph"
      },
      {
        "healthy": true,
        "name": "redis cache"
      },
      {
        "healthy": true,
        "name": "cassandra db"
      }
    ],
    "healthy": true
  }
}


Still I am getting the below error while trying to create a new framework:
**Response:**
{
    "id": "api.framework.create",
    "ver": "1.0",
    "ts": "2023-04-25T08:46:32.600Z",
    "params": {
        "resmsgid": "adc57180-e345-11ed-af80-afa5f3a40bf4",
        "msgid": null,
        "status": "failed",
        "err": null,
        "errmsg": null
    },
    "responseCode": "SERVER_ERROR",
    "result": {}
}

**Request:**
{
    "request": {
        "framework": {
            "name": "NCF Demo",
            "code": "ncfdemo",
            "description": "NCF Demo Framework"
        }
    }
}

**URL:** https://<domain_name>/api/framework/v1/create

**Reply by vinukumar-vs** [reply]
The logs from the taxonomy service will help to know what is the reason for failure. Verify the taxonomy service console logs & check the API is hitting to the service & is there any error logs in the service.

**Reply by ruchinksinha** [reply]
@vinukumar-vs There are no error in the taxonomy service. I believe the API call in not reaching the taxonomy service.

**Comment by ruchinksinha** [comment]
@rajeevsathish @chitranshu-keshav @Krishnaj20 @vinukumar-vs Looking forward for your support on this.

**Reply by vinukumar-vs** [reply]
@ruchinksinha I would suggest to discuss Framework related discussions in Knowlg discussion forum, because it is part of Knowlg BB.
https://github.com/orgs/Sunbird-Knowlg/discussions

We have documented the sequence of API's to be called to create the framework. Please refer the below
https://documenter.getpostman.com/view/25186239/2s93eR4vF1#3b6f599a-dfba-434c-aa5c-8515982d2989

[@Krishnaj20](https://github.com/Krishnaj20) [@pallakartheekreddy](https://github.com/pallakartheekreddy) @chitranshu-keshav @rajeevsathish
